test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -1189,34 +1186,61 @@
   (testing "Netscape comment constant has expected value"
     (is (= "Puppet Server Internal Certificate" netscape-comment-value))))
 
+(deftest ensure-no-authorization-extensions!-test
+  (testing "when checking a csr for authorization extensions"
+    (let [subject-keys (utils/generate-key-pair 512)
+          subject-pub  (utils/get-public-key subject-keys)
+          subject      "borges"
+          subject-dn   (utils/cn subject)
+
+          pp-auth-ext {:oid (:pp_authorization puppet-short-names)
+                       :value "true"
+                       :critical false}
+          pp-auth-role {:oid (:pp_auth_role puppet-short-names)
+                        :value "com"
+                        :critical false}
+          auth-csr (utils/generate-certificate-request subject-keys subject-dn [pp-auth-ext])
+          auth-role-csr (utils/generate-certificate-request subject-keys subject-dn [pp-auth-role])]
+
+      (testing "pp_authorization is caught"
+        (is (thrown+-with-msg?
+             [:kind :disallowed-extension]
+              #".*contains an authorization extension.*borges.*"
```

Comment:
I really like regexes that contain the minimum amount of a message that uniquely identifies it - or at least as much as I can bring myself to like a regex.   +1